src/tests/core/txs/test_wallet_pool.c

```diff
@@ -0,0 +1,220 @@
+#include <stdio.h>
+#include "minunit.h"
+#include "wallet_pool.h"
+#include "sign_tx.h"
+
+int tests_run = 0;
+
+Transaction *_make_tx() {
+  Transaction *tx;
+  Input *in;
+  Output *out;
+  mbedtls_ecdsa_context *key_pair;
+
+  tx = malloc(sizeof(Transaction));
+  in = malloc(sizeof(Input));
+  out = malloc(sizeof(Output));
+
+  memset(in, 0x25, sizeof(Input));
+  key_pair = gen_keys();
+  in->pub_key = &(key_pair->private_Q);
+
+  memset(out, 0x50, sizeof(Output));
+
+  tx->num_inputs = 1;
+  tx->inputs = in;
+  tx->num_outputs = 1;
+  tx->outputs = out;
+
+  return tx;
+}
+
+void _free_tx(Transaction *tx) {
+  free(tx->inputs);
+  free(tx->outputs);
+  free(tx);
+}
+
+static char *test_wallet_init() {
+  wallet_init();
+  mu_assert(
+      "Wallet pool was not initialized",
+      wallet_pool == NULL
+  );
+  mu_assert(
+      "Key pool was not initialized",
+      key_pool == NULL
+  );
+  return NULL;
+}
+
+static char  *test_wallet_pool_add() {
+  Transaction *tx;
+  WalletEntry *entry;
+  mbedtls_ecdsa_context *key_pair;
+
+  tx = _make_tx();
+  key_pair = gen_keys();
+
+  wallet_init();
+
+  entry = wallet_pool_add(tx, 0, key_pair);
+  mu_assert(
+    "Wallet entry amt incorrect",
+    entry->amt == tx->outputs[0].amt
+  );
+  mu_assert(
+    "Wallet key pair incorrect",
+    entry->key_pair == key_pair
+  );
+  mu_assert(
+    "Wallet entry spent incorrect",
+    entry->spent == 0
+  );
+
+  _free_tx(tx);
+  mbedtls_ecp_keypair_free(key_pair);
+  free(entry);
+
+  return NULL;
+}
+
+static char  *test_wallet_pool_find() {
+  Transaction *tx;
+  WalletEntry *entry, *ret_entry;
+  mbedtls_ecdsa_context *key_pair;
+  unsigned char hash[TX_HASH_LEN];
```

Comment:
in our case this is a transaction hash that actually has signatures filled in correct? is thi problematic?

Reply:
Signature being filled matters when doing verification since the signature is produced using the hash of a transaction with blank signature fields. In all other contexts we should assume signature fields are full